packages/website/components/account/fileUploader/fileUploader.js

```diff
@@ -95,8 +112,23 @@ const FileUploader = ({ className = '', content, uploadModalState, background })
           </div>
           <h5>{!!filesToUpload.length ? content.heading.option_1 : content.heading.option_2}</h5>
           <div className={'file-upload-subheading'} dangerouslySetInnerHTML={{ __html: content.subheading }}></div>
+
+          {!hasPaymentMethod && currentPlan?.id === 'free' && (
+            <div className="billing-terms-toggle">
+              <input type="checkbox" id="agreeTerms" checked={hasAcceptedTerms} onChange={e => acceptedTerms(e)} />
+              <label htmlFor="agreeTerms">
+                I have read and agree to the{' '}
+                <a href="/terms/" rel="noreferrer" target="_blank">
+                  web3.storage Terms of Service
+                </a>
+              </label>
+            </div>
+          )}
+
           <Dropzone
-            className="file-uploader-dropzone"
+            className={`file-uploader-dropzone ${
+              !hasPaymentMethod && currentPlan?.id === 'free' && !hasAcceptedTerms && 'disabled'
```

Comment:
not a blocker at all, but when conditional chains get this long I like to pull them out into local vars, e.g. `const dropZoneEnabled = thing && otherThing`.